semantic_router/tokenizers.py

```diff
@@ -0,0 +1,181 @@
+
+        :param path: Path to save the tokenizer to
+        :type path: str, :class:`pathlib.Path`
+        """
+        if isinstance(path, str):
+            path = Path(path)
+
+        with open(path, "w") as fp:
+            json.dump(self.config, fp)
+
+    @classmethod
+    def load(cls, path: str | Path) -> "BaseTokenizer":
+        """Returns a :class:`bm25_engine.tokenizer.BaseTokenizer` object from saved configuration
+
+        Requires these files:
+        - tokenizer.json: saved configuration of the tokenizer
+
+        :param path: Path to load the tokenizer from
+        :type path: str, :class:`pathlib.Path`
+        :returns: Configured BaseTokenizer
+        :rtype: BaseTokenizer
+        """
+        if isinstance(path, str):
+            path = Path(path)
+
+        with open(path) as fp:
+            config = json.load(fp)
+        return cls(**config)
+
+    def tokenize(self, texts: str | list[str], pad: bool = True) -> np.ndarray:
+        raise NotImplementedError
+
+
+class HashTokenizer(BaseTokenizer):
+    def __init__(self) -> None:
+        super().__init__()
+
+
+class PretrainedTokenizer(BaseTokenizer):
+    """Wrapper for HuggingFace tokenizers, representing a pretrained tokenizer (i.e. bert-base-uncased).
+    Extends the :class:`semantic_router.tokenizers.BaseTokenizer` class.
+
+    :param tokenizer: Binding for HuggingFace Rust tokenizers
+    :type tokenizer: class:`tokenizers.Tokenizer`
+    :param add_special_tokens: Whether to accept special tokens from the tokenizer (i.e. `[PAD]`)
+    :type add_special_tokens: bool
+    :param pad: Whether to pad the input to a consistent length (using `[PAD]` tokens)
+    :type pad: bool
+    :param model_ident: HuggingFace ID of the model (i.e. `bert-base-uncased`)
+    :type model_ident: str
+    """
+
+    tokenizer: Tokenizer
+    add_special_tokens: bool
+    pad: bool
+    model_ident: str
+
+    def __init__(
+        self,
+        model_ident: str,
+        custom_normalizer: Sequence | None = None,
+        add_special_tokens: bool = False,
+        pad: bool = True,
+    ) -> None:
+        """Constructor method"""
+        super().__init__()
+        self.add_special_tokens = add_special_tokens
+        self.model_ident = model_ident
+        self.tokenizer = self._initialize_hf_tokenizers()
+        self.pad = pad
+        if custom_normalizer:
+            self.tokenizer.normalizer = custom_normalizer  # type: ignore
+        if pad:
+            self.tokenizer.enable_padding(direction="right", pad_id=0)
+
+    def _initialize_hf_tokenizers(self) -> Tokenizer:
+        try:
+            from tokenizers import Tokenizer
+        except ImportError:
+            raise ImportError(
+                "Please install tokenizers to use PretrainedTokenizer. "
+                "You can install it with: "
+                "`pip install semantic-router[hybrid]`"
+            )
+        else:
+            return Tokenizer.from_pretrained(self.model_ident)
```

Comment:
```suggestion
        return Tokenizer.from_pretrained(self.model_ident)
```

no need for the try-except, we don't have `hybrid` anymore and `tokenizers` is installed by default